tools/query_verification/src/main/java/com/google/bigquery/QueryVerifier.java

```diff
@@ -38,7 +44,49 @@ public void verify() {
     public void verifyDataFree() {
         boolean verificationResult = false;
 
-        // TODO Implement data free verification
+        BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
+
+        List<Table> tables = new ArrayList<Table>();
+
+        // Create tables based on schema
+        if (migratedSchema != null) {
+            if (migratedSchema.isInJsonFormat()) {
+                TableInfo tableInfo = QueryVerifier.getTableInfoFromJsonSchema(migratedSchema);
+                if (tableInfo != null) {
+                    Table table = bigQuery.create(tableInfo);
+                    tables.add(table);
+                } else {
+                    System.out.println(migratedSchema.path() + " is not correctly formatted.");
+                }
+            } else {
+                // TODO Load schema from DDL
+            }
+        }
+
+        // Create dry-run job
+        QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(migratedQuery.query())
+                .setDryRun(true)
+                .build();
+        JobId jobId = JobId.of(UUID.randomUUID().toString());
+        JobInfo jobInfo = JobInfo.newBuilder(queryConfig)
+                .setJobId(jobId)
+                .build();
+
+        // TODO Support multiple queries
+
+        try {
+            // Run dry-run
+            Job queryJob = bigQuery.create(jobInfo);
+            verificationResult = queryJob.getStatus().getState() == JobStatus.State.DONE;
+        } catch (BigQueryException e) {
+            // Print out syntax/semantic errors returned from BQ
```

Comment:
Have you tested an exception is thrown due to a bad query, not just due to something like a failure to connect/bad credentials etc?

Reply:
I've found that if the query is invalid from a syntax or semantic error, BQ throws an exception so I can just print the error message in catch block.

Reply:
Ok that should be good for now then, thanks!